Source/Scene/Cesium3DTileFeatureTableResources.js

```diff
@@ -0,0 +1,82 @@
+/*global define*/
+define([
+        '../Core/ComponentDataType',
+        '../Core/defaultValue',
+        '../Core/defined',
+        '../Core/DeveloperError'
+    ], function(
+        ComponentDataType,
+        defaultValue,
+        defined,
+        DeveloperError) {
+    'use strict';
+
+    /**
+     * @private
+     */
+    function Cesium3DTileFeatureTableResources(featureTableJSON, featureTableBinary) {
+        this.json = featureTableJSON;
+        this.buffer = featureTableBinary;
+        this._cachedArrayBufferViews = {};
+        this.featuresLength = 0;
+    }
+
+    Cesium3DTileFeatureTableResources.prototype.getTypedArrayForSemantic = function(semantic, byteOffset, componentType, count, featureSize) {
+        //>>includeStart('debug', pragmas.debug);
+        if (!defined(byteOffset)) {
+            throw new DeveloperError('byteOffset must be defined to read from binary data for semantic: ' + semantic);
+        }
+        if (!defined(componentType)) {
+            throw new DeveloperError('componentType must be defined to read from binary data for semantic: ' + semantic);
+        }
+        if (!defined(count)) {
+            throw new DeveloperError('count must be defined to read from binary data for semantic: ' + semantic);
+        }
+        //>>includeEnd('debug');
+        var cachedArrayBufferViews = this._cachedArrayBufferViews;
+        var arrayBuffer = cachedArrayBufferViews[semantic];
+        if (!defined(arrayBuffer)) {
+            arrayBuffer = ComponentDataType.createArrayBufferView(componentType, this.buffer.buffer, this.buffer.byteOffset + byteOffset, count * featureSize);
+            cachedArrayBufferViews[semantic] = arrayBuffer;
+        }
+        return arrayBuffer;
+    };
+
+    Cesium3DTileFeatureTableResources.prototype.getGlobalProperty = function(semantic, componentType, count) {
+        var jsonValue = this.json[semantic];
+        if (defined(jsonValue)) {
+            var byteOffset = jsonValue.byteOffset;
+            if (defined(byteOffset)) {
+                // This is a reference to the binary
+                count = defaultValue(count, 1);
+                var typedArray = this.getTypedArrayForSemantic(semantic, byteOffset, componentType, count);
+                var subArray = typedArray.subarray(0, count);
+                if (count === 1) {
```

Comment:
What is the use case for implicitly converting an array of length 1 to a scalar value?  Why wouldn't we just presenter this to the user as an array like is stored in the tile.